src/main/java/com/synopsys/integration/detect/lifecycle/boot/product/BlackDuckConnectivityChecker.java

```diff
@@ -70,6 +75,8 @@ public BlackDuckConnectivityResult determineConnectivity(BlackDuckServerConfig b
             );
         }
 
-        return BlackDuckConnectivityResult.success(blackDuckServicesFactory, blackDuckServerConfig);
+        BlackDuckConnectivityResult result =  BlackDuckConnectivityResult.success(blackDuckServicesFactory, blackDuckServerConfig);
+        result.setContactedServerVersion(version);
+        return result;
```

Comment:
Rather than taking BlackDuckConnectivityResult from immutable to mutable, why not keep it mutable and enhance the static factory methods it has to accept version?

Reply:
I was concerned I'd be messing up signatures in areas which did not care about the version string....

Reply:
Done